drivers/gles3/storage/render_scene_buffers_gles3.cpp

```diff
@@ -31,30 +31,508 @@
+		WARN_PRINT_ONCE("MSAA is not supported on this device.");
+		msaa3d.mode = RS::VIEWPORT_MSAA_DISABLED;
+	} else if (use_multiview && msaa3d.mode != RS::VIEWPORT_MSAA_DISABLED && !config->msaa_multiview_supported && !config->rt_msaa_multiview_supported) {
+		WARN_PRINT_ONCE("Multiview MSAA is not supported on this device.");
+		msaa3d.mode = RS::VIEWPORT_MSAA_DISABLED;
+	}
+
+	if (msaa3d.mode != RS::VIEWPORT_MSAA_DISABLED) {
+		// Setup MSAA.
+		const GLsizei samples[] = { 1, 2, 4, 8 };
+		msaa3d.samples = samples[msaa3d.mode];
+
+		// Constrain by limits of OpenGL driver.
+		if (msaa3d.samples > config->msaa_max_samples) {
+			msaa3d.samples = config->msaa_max_samples;
+		}
+
+		if (!use_multiview && !config->rt_msaa_supported) {
+			// Render to texture extensions not supported? fall back to MSAA framebuffer through GL_EXT_framebuffer_multisample.
+			// Note, if 2D MSAA matches 3D MSAA and we're not scaling, it would be ideal if we reuse our 2D MSAA buffer here.
+			// We can't however because we don't trigger a change in configuration if 2D MSAA changes.
+			// We'll accept the overhead in this situation.
+
+			msaa3d.needs_resolve = true;
+			msaa3d.check_fbo_cache = false;
+
+			bool is_transparent = texture_storage->render_target_get_transparent(render_target);
+			GLuint color_internal_format = is_transparent ? GL_RGBA8 : GL_RGB10_A2;
+
+			// Create our color buffer.
+			glGenRenderbuffers(1, &msaa3d.color);
+			glBindRenderbuffer(GL_RENDERBUFFER, msaa3d.color);
+
+			glRenderbufferStorageMultisample(GL_RENDERBUFFER, msaa3d.samples, color_internal_format, internal_size.x, internal_size.y);
+			GLES3::Utilities::get_singleton()->render_buffer_allocated_data(msaa3d.color, internal_size.x * internal_size.y * view_count * 4 * msaa3d.samples, "MSAA 3D color render buffer");
+
+			// Create our depth buffer.
+			glGenRenderbuffers(1, &msaa3d.depth);
+			glBindRenderbuffer(GL_RENDERBUFFER, msaa3d.depth);
+
+			glRenderbufferStorageMultisample(GL_RENDERBUFFER, msaa3d.samples, GL_DEPTH_COMPONENT24, internal_size.x, internal_size.y);
+			GLES3::Utilities::get_singleton()->render_buffer_allocated_data(msaa3d.depth, internal_size.x * internal_size.y * view_count * 3 * msaa3d.samples, "MSAA 3D depth render buffer");
+
+			// Create our MSAA 3D FBO.
+			glGenFramebuffers(1, &msaa3d.fbo);
+			glBindFramebuffer(GL_FRAMEBUFFER, msaa3d.fbo);
+
+			glFramebufferRenderbuffer(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_RENDERBUFFER, msaa3d.color);
+			glFramebufferRenderbuffer(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_RENDERBUFFER, msaa3d.depth);
+
+			GLenum status = glCheckFramebufferStatus(GL_FRAMEBUFFER);
+			if (status != GL_FRAMEBUFFER_COMPLETE) {
+				_clear_msaa3d_buffers();
+				WARN_PRINT("Could not create 3D MSAA buffers, status: " + texture_storage->get_framebuffer_error(status));
+			}
+
+			glBindTexture(GL_RENDERBUFFER, 0);
```

Comment:
```suggestion
			glBindTexture(GL_TEXTURE_2D, 0);
```
I'm getting the following error:

```
ERROR: GL ERROR: Source: OpenGL	Type: Error	ID: 1	Severity: High	Message: GL_INVALID_ENUM in glBindTexture(target = GL_RENDERBUFFER)
```
This change fixes it

Reply:
Ah, it's actually `glBindRenderbuffer(GL_RENDERBUFFER, 0);`

This is the only place we use render buffers instead of textures :)